src/server/svr_jobfunc.c

```diff
@@ -4603,35 +4603,43 @@ start_end_dur_wall(void *pobj, int objtype)
 			return (-1);
 	}
 	else {
-		swcode = 3;
+		if (presv->ri_alter_flags & RESV_DURATION_MODIFIED) {
+			if (presv->ri_alter_flags & RESV_END_TIME_MODIFIED)
+				swcode += 6; /*calcualte start time*/
+			if (presv->ri_alter_flags & RESV_START_TIME_MODIFIED)
+				swcode += 5; /*calculate end time*/;
+			if (swcode == 11)
+				return (-1);
+		}
+		else {
+			swcode = 3; /*start, end*/
+		}
```

Comment:
I would change this to something like:
```C
if (presv->ri_alter_flags & RESV_START_TIME_MODIFIED)
    swcode += 1;
if (flags & end_modified)
    += 2;
if (flags & duration_modified)
    += 4;
```
Here you would also have to check if duration is the only thing passed, to use duration.
```C
if (flags == duration_modified)
    swcode += 1;
```

Then in the switch-case statement below, add cases for 1, 2, and 4, rejecting them.

Reply:
I changed it according to first part of comment. 
For the second part I just combined case 1 and 2 similar to case 3 and case 4 similar to case 5.